### PR TITLE
Implementing tab click response

### PR DESCRIPTION
**Pull request by coldrohak**
#13 
@fliptrail i tried implementing the feature but setImageCounter(selectedValue) is showing error as type of selectedValue is unknown, if you can help out, the feature is working though..

### CODE REVIEW

File: src/app/components/carousel.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 "use client";
 
-import { useEffect, useState } from "react";
+import { use, useEffect, useState } from "react";
```
Comment (fliptrail) [inline src/app/components/carousel.tsx:3]: What is this change?

File: src/app/components/carousel.tsx [hunk 2]
```diff
@@ -16,16 +16,27 @@ import {
...
+  const [selectedValue, setSelectedValue] = useState<TabValue>(0);
 
   useEffect(() => {
     setImageSrc(imageArray[imageCounter]);
+    setSelectedValue(imageCounter);
   }, [imageCounter]);
 
+  useEffect(() => {
+    setImageCounter(selectedValue);
+  }, [selectedValue]);
```
Comment (fliptrail) [inline src/app/components/carousel.tsx:38]: Check datatype here

**Review by fliptrail — CHANGES_REQUESTED** [review]

File: src/app/components/carousel.tsx [hunk 3]
```diff
@@ -16,16 +16,27 @@ import {
...
+  SelectTabData,
+  SelectTabEvent,
+  TabValue,
+} from "@fluentui/react-components";
 
 const Carousel = () => {
   const [imageCounter, setImageCounter] = useState(0);
   const imageArray = ["/cities_pixel.jpg", "/monopoly.jpeg", "/business.jpeg"];
   const [imageSrc, setImageSrc] = useState(imageArray[0]);
+  const [selectedValue, setSelectedValue] = useState<TabValue>(0);
```
Comment (fliptrail) [inline src/app/components/carousel.tsx:29]: replace with `useState<number>(0)`

File: src/app/components/carousel.tsx [hunk 4]
```diff
@@ -38,14 +49,18 @@ const Carousel = () => {
     });
   };
 
+  const onTabSelect = (event: SelectTabEvent, data: SelectTabData) => {
+    setSelectedValue(data.value);
```
Comment (fliptrail) [inline src/app/components/carousel.tsx:53]: replace with `setSelectedValue(data.value as number);`

**Review by fliptrail — CHANGES_REQUESTED** [review]

**Comment by coldrohak** [comment]
@fliptrail thanks for help, is this fine now?

**Comment by fliptrail** [comment]
LGTM! Thank you for your contribution.

**Pull request merged**